easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -918,6 +939,33 @@ def resolve_template(value, tmpl_dict):
 
     return value
 
+def replace_templates(value, templ_const, templ_val):
+    """
+    Given a value, try to substitute template strings where possible.
+        - value can be a string, list, tuple, dict or combination thereof
+        - templ_const is a dictionary of template strings (constants)
+        - templ_val is an ordered dictionary of template strings specific for this easyconfig file
+    """
+    if isinstance(value, basestring):
+        old_value = None
+        while value != old_value:
+            old_value = value
+            if value in templ_const:
+                value = templ_const[value]
+            else:
+                # check for template values
+                for k, v in templ_val.items():
```

Comment:
don't use single-letter variables outside of a list comprehension, `key, val` is fine